cli/src/commands/deps.rs

```diff
@@ -0,0 +1,261 @@
+        graph.edge(ident, unknown);
+    }
+}
+
+fn check_expr<'a>(
+    expr: &'a Expr<'a>,
+    rule_name: &'a str,
+    dep_map: &mut BTreeMap<&'a str, Deps<'a>>,
+) {
+    let mut variables: Vec<&str> = vec![];
+    for event in DFSIter::new(expr) {
+        match event {
+            DFSEvent::Enter(expr) => match expr {
+                Expr::Ident(ident) => {
+                    if let Some(_) = dep_map.get(ident.name) {
+                        // This is an identifier that matches a previously seen
+                        // rule.
+                        dep_map.entry(rule_name).and_modify(|v| {
+                            v.rules.insert(ident.name);
+                        });
+                    } else if yara_x::mods::module_names()
+                        .any(|module| module == ident.name)
+                        && !variables.contains(&ident.name)
+                    {
+                        // This is a known module or is not in the list of
+                        // variable identifier to be ignored.
+                        dep_map.entry(rule_name).and_modify(|v| {
+                            v.modules.insert(ident.name);
+                        });
+                    } else {
+                        dep_map.entry(rule_name).and_modify(|v| {
+                            v.unknowns.insert(ident.name);
+                        });
+                    }
+                }
+                Expr::Of(of) => match &of.quantifier {
+                    Quantifier::Percentage(quantifier)
+                    | Quantifier::Expr(quantifier) => {
+                        check_expr(&quantifier, rule_name, dep_map);
+                    }
+                    _ => {}
+                },
+                Expr::ForOf(for_of) => match &for_of.quantifier {
+                    Quantifier::Percentage(quantifier)
+                    | Quantifier::Expr(quantifier) => {
+                        check_expr(&quantifier, rule_name, dep_map);
+                    }
+                    _ => {}
+                },
+                Expr::ForIn(for_in) => {
+                    variables =
+                        for_in.variables.iter().map(|v| v.name).collect();
+                    match &for_in.quantifier {
+                        Quantifier::Percentage(quantifier)
+                        | Quantifier::Expr(quantifier) => {
+                            check_expr(&quantifier, rule_name, dep_map);
+                        }
+                        _ => {}
+                    }
+                }
+                Expr::With(with) => {
+                    variables = with
+                        .declarations
+                        .iter()
+                        .map(|d| d.identifier.name)
+                        .collect();
+                }
+                _ => {}
+            },
+            DFSEvent::Leave(expr) => match expr {
+                Expr::ForIn(_) | Expr::With(_) => {
+                    variables.clear();
```

Comment:
This won't work nested loops o with statements, once you exit the inner loop/with you are clearing all the variables, including the ones declared in outter ones.